applications/nrf_desktop/pytest/test_mcuboot_kmu.py

```diff
@@ -0,0 +1,103 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+import os
+import logging
+import subprocess
+import re
+
+from twister_harness import DeviceAdapter
+from twister_harness.helpers.utils import find_in_config
+
+logger = logging.getLogger(__name__)
+
+
+def run_command(command: list[str], timeout: int = 60):
+    ret: subprocess.CompletedProcess = subprocess.run(command,
+                                                      text=True,
+                                                      stdout=subprocess.PIPE,
+                                                      stderr=subprocess.STDOUT,
+                                                      timeout=timeout)
+    if ret.returncode:
+        logger.error("Command failed: {}".format(" ".join(command)))
+        logger.error(ret.stdout)
+        raise subprocess.CalledProcessError(ret.returncode, command)
+
+
+def board_erase(dut: DeviceAdapter):
+    command = ["nrfutil", "device", "erase"]
+    if dut.device_config.id:
+        command.extend(["--serial-number", dut.device_config.id])
+
+    logger.info("Erasing board")
+    run_command(command)
+
+
+def _mcuboot_key_path(dut: DeviceAdapter):
+    mcuboot_conf_path = os.path.join(str(dut.device_config.build_dir), "mcuboot", "zephyr",
+                                     ".config")
+    return find_in_config(mcuboot_conf_path, "CONFIG_BOOT_SIGNATURE_KEY_FILE")
+
+
+def mcuboot_provision(dut: DeviceAdapter):
+    soc = dut.device_config.platform.split("/")[1]
+    key_path = _mcuboot_key_path(dut)
+    command = ["west", "ncs-provision", "upload", "-s", soc, "-k", key_path]
+    if dut.device_config.id:
+        command.extend(["--dev-id", dut.device_config.id])
+
+    logger.info("KMU provisioning")
+    run_command(command)
+
+
+def board_program(dut: DeviceAdapter):
+    command = ["west", "flash", "--skip-rebuild", "-d", str(dut.device_config.build_dir)]
+    if dut.device_config.id:
+        command.extend(["--dev-id", dut.device_config.id])
+
+    logger.info("Programming board")
+    run_command(command)
+
+
+def logs_verify(dut: DeviceAdapter):
+    # Expected logs are sourced from nRF Desktop `sample.yaml` file.
+    expected_logs = [
+      "app_event_manager: e:module_state_event module:main state:READY",
+      "ble_state: Bluetooth initialized",
+      "settings_loader: Settings loaded",
+      "ble_bond: Selected Bluetooth LE peers",
+      "(ble_adv: Advertising started)|(ble_scan: Scan started)",
+      "dfu: Secondary image slot is clean"
+    ]
+
+    expected_regexes = list(map(re.compile, expected_logs))
+
+    while True:
+        line = dut.readline(timeout=120)
```

Comment:
I prefer to avoid dependency on log order (the log order is not guaranteed). I will add early return on an error log. Prefer to keep timeout longer if possible to ensure that device will have necessary time to erase the secondary image slot in the background.